internal/lookup/root/root.go

```diff
@@ -53,6 +55,24 @@ func (r *Driver) Libraries() lookup.Locator {
 	)
 }
 
+// Configs returns a locator for driver configs.
+// If configSearchPaths is specified, these paths are used as absolute paths,
+// otherwise, /etc and /usr/share are searched.
+func (r *Driver) Configs() lookup.Locator {
+	useRoot := r.Root
+	searchPaths := []string{"/etc", "/usr/share"}
+	if len(r.configSearchPaths) > 0 {
+		useRoot = ""
```

Comment:
Instead of setting this as an empty string, I would prefer if we added the with `.WithRoot()` to the Options list when configSearchPaths is not set

Reply:
I took a stab at reworking this as:
```
diff --git a/internal/lookup/root/root.go b/internal/lookup/root/root.go
index bfd3ace4..f3176744 100644
--- a/internal/lookup/root/root.go
+++ b/internal/lookup/root/root.go
@@ -59,17 +59,22 @@ func (r *Driver) Libraries() lookup.Locator {
 // If configSearchPaths is specified, these paths are used as absolute paths,
 // otherwise, /etc and /usr/share are searched.
 func (r *Driver) Configs() lookup.Locator {
-	useRoot := r.Root
-	searchPaths := []string{"/etc", "/usr/share"}
-	if len(r.configSearchPaths) > 0 {
-		useRoot = ""
-		searchPaths = normalizeSearchPaths(r.configSearchPaths...)
+	options := []lookup.Option{
+		lookup.WithLogger(r.logger),
 	}
 
+	if len(r.configSearchPaths) == 0 {
+		options = append(options,
+			lookup.WithRoot(r.Root),
+			lookup.WithSearchPaths("/etc", "/usr/share"),
+		)
+	} else {
+		options = append(options,
+			lookup.WithSearchPaths(normalizeSearchPaths(r.configSearchPaths...)...),
+		)
+	}
 	return lookup.NewFileLocator(
-		lookup.WithLogger(r.logger),
-		lookup.WithRoot(useRoot),
-		lookup.WithSearchPaths(searchPaths...),
+		options...,
 	)
 }
 
```
but I find this more verbose and unclear as to what is actually being done here. If you don't mind, I will leave this as is -- possibly renaming some local variables.